runtime/src/asset.rs

```diff
@@ -961,18 +963,22 @@ decl_module! {
             ticker.canonize();
             ensure!(Self::is_owner(&ticker, did), "user is not authorized");
 
-            <Documents<T>>::insert((ticker, name), (uri, document_hash, <pallet_timestamp::Module<T>>::get()));
-            Ok(())
+            for doc in documents.iter() {
+                Self::add_document(ticker, doc.clone());
+            }
         }
 
-        /// Used to remove the document details for the given token, Only be called by the token owner
+        /// Update document details for the given token, Only be called by the token owner
         ///
         /// # Arguments
         /// * `origin` Signing key of the token owner
         /// * `did` DID of the token owner
         /// * `ticker` Ticker of the token
-        /// * `name` Name of the document
-        pub fn remove_document(origin, did: IdentityId, ticker: Ticker, name: Vec<u8>) -> DispatchResult {
+        /// * `doc_id` Document id
+        /// * `name` Document name
+        /// * `uri` Document URI
+        /// * `document_hash` Document hash
+        pub fn update_document(origin, did: IdentityId, ticker: Ticker, doc_id: u64, name: Vec<u8>, uri: Vec<u8>, document_hash: Vec<u8>) -> DispatchResult {
```

Comment:
Since we already have the struct, it will be cleaner to take doc details as struct input. i.e. `document: Document` rather than individual name, uri and hash.

Reply:
Updated signature to take document instead.